api/src/controllers/login.js

```diff
@@ -229,10 +251,64 @@ const getBranding = () => {
     });
 };
 
+const renderTokenLogin = (req, res) => {
+  return getBranding()
+    .then(branding => render('tokenLogin', req, branding, { tokenUrl: req.url }))
+    .then(body => res.send(body));
+};
+
+const createSessionRetry = (req, retry= 10) => {
+  return createSession(req).then(sessionRes => {
+    if (sessionRes.statusCode === 200) {
+      return sessionRes;
+    }
+
+    if (retry) {
```

Comment:
It's paranoid of me but I prefer...

```suggestion
    if (retry >= 0) {
```

Otherwise if we accidentally decrement twice and skip over `0` it'll infinite loop...